Bottom: 5d4fa64b933a9ebba465845e5cda2302abaeb606
Top:    a06b284f142de4fe49fcec3f72f897fe767c44ac
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2014-06-28 21:16:01 +0100

Remove PreLoadElementPCB and PostLoadElementPCB

Where these are called from parse_y.y, yyPCB is always NULL, and
the functions do nothing.


---

diff --git a/src/file.c b/src/file.c
index ca22071..b775961 100644
--- a/src/file.c
+++ b/src/file.c
@@ -84,7 +84,6 @@
 #include "file.h"
 #include "hid.h"
 #include "misc.h"
-#include "move.h"
 #include "mymem.h"
 #include "parse_l.h"
 #include "pcb-printf.h"
@@ -495,48 +494,6 @@ RevertPCB (void)
 }
 
 /* ---------------------------------------------------------------------------
- * functions for loading elements-as-pcb
- */
-
-extern	PCBType *		yyPCB;
-extern	DataType *		yyData;
-extern	FontType *		yyFont;
-
-void
-PreLoadElementPCB ()
-{
-
-  if (!yyPCB)
-    return;
-
-  yyFont = &yyPCB->Font;
-  yyData = yyPCB->Data;
-  yyData->pcb = yyPCB;
-  yyData->LayerN = 0;
-}
-
-void
-PostLoadElementPCB ()
-{
-  PCBType *pcb_save = PCB;
-  ElementType *e;
-
-  if (!yyPCB)
-    return;
-
-  CreateNewPCBPost (yyPCB, 0);
-  ParseGroupString("1,c:2,s", &yyPCB->LayerGroups, yyData->LayerN);
-  e = yyPCB->Data->Element->data; /* we know there's only one */
-  PCB = yyPCB;
-  MoveElementLowLevel (yyPCB->Data,
-		       e, -e->BoundingBox.X1, -e->BoundingBox.Y1);
-  PCB = pcb_save;
-  yyPCB->MaxWidth = e->BoundingBox.X2;
-  yyPCB->MaxHeight = e->BoundingBox.Y2;
-  yyPCB->is_footprint = 1;
-}
-
-/* ---------------------------------------------------------------------------
  * writes the quoted string created by another subroutine
  */
 static void
diff --git a/src/file.h b/src/file.h
index 4056bc9..fd56f90 100644
--- a/src/file.h
+++ b/src/file.h
@@ -46,8 +46,6 @@ void DisableEmergencySave (void);
 int ReadLibraryContents (void);
 int ImportNetlist (char *);
 int SaveBufferElements (char *);
-void PreLoadElementPCB (void);
-void PostLoadElementPCB (void);
 void sort_netlist (void);
 
 /* 
diff --git a/src/parse_y.y b/src/parse_y.y
index 72a0046..4765ccf 100644
--- a/src/parse_y.y
+++ b/src/parse_y.y
@@ -54,6 +54,7 @@
 #include "rtree.h"
 #include "strflags.h"
 #include "thermal.h"
+#include "move.h"
 
 #ifdef HAVE_LIBDMALLOC
 # include <dmalloc.h> /* see http://dmalloc.com */
@@ -213,13 +214,45 @@ parsepcb
 			PCB = pcb_save;
 			}
 			   
-		| { PreLoadElementPCB ();
-		    layer_group_string = NULL; }
+		| {
+
+		    if (yyPCB != NULL)
+		      {
+                        printf ("Begin parsing element...");
+			yyFont = &yyPCB->Font;
+			yyData = yyPCB->Data;
+			yyData->pcb = yyPCB;
+			yyData->LayerN = 0;
+			layer_group_string = NULL;
+		      }
+                    else
+                      printf ("yyPCB=NULL Begin parsing element...");
+		  }
 		  element
-		  { LayerFlag[0] = true;
-		    LayerFlag[1] = true;
-		    yyData->LayerN = 2;
-		    PostLoadElementPCB ();
+		  {
+		    PCBType *pcb_save = PCB;
+		    ElementType *e;
+
+		    if (yyPCB != NULL)
+		      {
+                        printf ("done\n");
+
+			LayerFlag[0] = true;
+			LayerFlag[1] = true;
+			yyData->LayerN = 2;
+
+			CreateNewPCBPost (yyPCB, 0);
+			ParseGroupString("1,c:2,s", &yyPCB->LayerGroups, yyData->LayerN);
+			e = yyPCB->Data->Element->data; /* we know there's only one */
+			PCB = yyPCB;
+			MoveElementLowLevel (yyPCB->Data, e, -e->BoundingBox.X1, -e->BoundingBox.Y1);
+			PCB = pcb_save;
+			yyPCB->MaxWidth = e->BoundingBox.X2;
+			yyPCB->MaxHeight = e->BoundingBox.Y2;
+			yyPCB->is_footprint = 1;
+		      }
+                    else
+                      printf ("done, BTW: yyData is %p\n", yyData);
 		  }
 		;
